modules/gapi/include/opencv2/gapi/oak/infer.hpp

```diff
@@ -0,0 +1,66 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2022 Intel Corporation
+
+#ifndef OPENCV_GAPI_OAK_INFER_HPP
+#define OPENCV_GAPI_OAK_INFER_HPP
+
+#include <unordered_map>
+#include <string>
+#include <array>
+#include <tuple>
+
+#include <opencv2/gapi/opencv_includes.hpp>
+#include <opencv2/gapi/util/any.hpp>
+
+#include <opencv2/core/cvdef.h>     // GAPI_EXPORTS
+#include <opencv2/gapi/gkernel.hpp> // GKernelPackage
+
+namespace cv {
+namespace gapi {
+namespace oak {
+
+namespace detail {
+/**
+* @brief This structure contains description of inference parameters
+* which is specific to OAK models.
+*/
+struct ParamDesc {
+    std::string blob_file;
```

Comment:
No input/output layer names here? How is this supposed to work? So far you test with 1-in 1-out models only?

Reply:
Currently only 1-in 1-out models are supported. Will extend once supported. Although, not sure if this info will be needed from the user since we extract all meta information from the model directly and utilize in the backend